packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -256,6 +328,209 @@ function toCollectorInstrumentationLibrarySpans(
+  } else if (
+    metricKind === MetricKind.UP_DOWN_COUNTER ||
+    metricKind === MetricKind.UP_DOWN_SUM_OBSERVER
+  ) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  } else if (
+    metricKind === MetricKind.VALUE_OBSERVER ||
+    metricKind === MetricKind.VALUE_RECORDER
+  ) {
+    // TODO: Change once LastValueAggregator is implemented.
+    // If the aggregator is LastValue or Exact, then it will be instantaneous
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  }
+  return opentelemetryProto.metrics.v1.MetricDescriptorTemporality
+    .INVALID_TEMPORALITY;
+}
+
+/**
+ * Given a MetricRecord, return the Collector compatible type of MetricDescriptor
+ * @param metric
+ */
+export function toCollectorMetricDescriptor(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptor {
+  return {
+    name: metric.descriptor.name,
+    description: metric.descriptor.description,
+    unit: metric.descriptor.unit,
+    type: toCollectorType(metric),
+    temporality: toCollectorTemporality(metric),
+  };
+}
+
+/**
+ * Returns an Int64Point or DoublePoint to the collector
+ * @param metric
+ * @param startTime
+ */
+export function toSingularPoint(
+  metric: MetricRecord,
+  startTime: number
+): {
+  labels: opentelemetryProto.common.v1.StringKeyValue[];
+  startTimeUnixNano: number;
+  timeUnixNano: number;
+  value: number;
+} {
+  let pointValue: number;
+  if (metric.aggregator instanceof MinMaxLastSumCountAggregator) {
+    pointValue = (metric.aggregator.toPoint().value as Distribution).last;
+  } else {
+    pointValue = metric.aggregator.toPoint().value as number;
+  }
+  return {
+    labels: toCollectorLabels(metric.labels),
+    value: pointValue,
+    startTimeUnixNano: startTime,
+    timeUnixNano: core.hrTimeToNanoseconds(
+      metric.aggregator.toPoint().timestamp
+    ),
+  };
+}
+
+/**
+ * Returns a HistogramPoint to the collector
+ * @param metric
+ * @param startTime
+ */
+export function toHistogramPoint(
+  metric: MetricRecord,
+  startTime: number
+): opentelemetryProto.metrics.v1.HistogramDataPoint {
+  const histValue = metric.aggregator.toPoint().value as Histogram;
+  return {
+    labels: toCollectorLabels(metric.labels),
+    sum: histValue.sum,
+    count: histValue.count,
+    startTimeUnixNano: startTime,
+    timeUnixNano: core.hrTimeToNanoseconds(
+      metric.aggregator.toPoint().timestamp
+    ),
+    buckets: histValue.buckets.counts.map(count => {
+      return { count };
+    }),
+    explicitBounds: histValue.buckets.boundaries,
+  };
+}
+
+/**
+ * Converts a metric to be compatible with the collector
+ * @param metric
+ * @param startTime start time in nanoseconds
+ */
+export function toCollectorMetric(
+  metric: MetricRecord,
+  startTime: number
+): opentelemetryProto.metrics.v1.Metric {
+  // All data points are set as empty by default
+  let int64DataPoints: opentelemetryProto.metrics.v1.Int64DataPoint[] = [];
```

Comment:
Why allocate this array just to overwrite it in a few lines? Is there a way to do this without `let`?